components/request-template.js

```diff
@@ -3,124 +3,110 @@
+var wrapper = require('../src/javascript-wrapper.js');
 
-exports.getComponent = componentFactory({
+/**
+ * Initials an HTTP request from uri-template (RFC6570), using object data from
+ * the @in port. Resolve 200/300 response body or rejects 400/500 response body
+ * @param method HTTP method
+ * @param url URI-Template (RFC6570)
+ * @param headers Request headers as name:template pairs using the URI-Template syntax
+ * @param body Request body template using the Handlebars syntax
+ * @param parameters Object data used to populate the templates, but not trigger the request
+ * @param input Object data used to populate the templates and trigger the request
+ */
+module.exports = wrapper({
     description: "Initials an HTTP request from uri-template (RFC6570),\
         using object data from the @in port. Resolve 200/300 response body or rejects 400/500 response body",
     icon: 'external-link',
-    outPorts: promiseOutput.outPorts,
     inPorts: {
         method: {
             description: "HTTP method",
-            datatype: 'string',
-            ondata: function(method) {
-                this.nodeInstance.method = newUriTemplate(method);
-            }
+            datatype: 'string'
         },
         url: {
             description: "URI-Template (RFC6570)",
-            datatype: 'string',
-            required: true,
-            ondata: function(url) {
-                this.nodeInstance.url = newUriTemplate(url);
-            }
+            datatype: 'string'
         },
         headers: {
             description: "Request headers as name:template pairs using the URI-Template syntax",
             datatype: 'object',
-            ondata: function(headers) {
-                this.nodeInstance.headers = newUriTemplate(headers);
-            }
+            multi: true
         },
         body: {
             description: "Request body template using the Handlebars syntax",
-            datatype: 'string',
-            ondata: function(body) {
-                this.nodeInstance.body = Handlebars.compile(body);
-            }
+            datatype: 'string'
         },
         parameters: {
             description: "Object data used to populate the templates, but not trigger the request",
             datatype: 'object',
-            ondata: function(parameters) {
-                this.nodeInstance.parameters = _.defaults(parameters, this.nodeInstance.parameters);
-            }
+            multi: true
         },
         input: {
             description: "Object data used to populate the templates and trigger the request",
-            datatype: 'object',
-            required: true,
-            ondata: function(data) {
-                this.nodeInstance.data = _.defaults(data, this.nodeInstance.data);
-            },
-            ondisconnect: promiseOutput(execute)
+            datatype: 'object'
         }
+    },
+    updater: function(method, url, headers, body, parameters, input) {
```

Comment:
For style I would rather have updaters be  named functions (instead of being defined inline as lambdas), so that they are more recognizable as separable functions, like this:
<pre>
module.exports = wrapper({
  . . . 
  updater: foo
});

function foo(a, b, c) {
  . . . 
}
</pre>